blog/2019-03-14-debezium-meets-quarkus.adoc

```diff
@@ -0,0 +1,193 @@
+--
+
+Quarkus is a Java stack designed for the development of cloud-native applications based on the Java platform.
+It combines and tightly integrates mature libraries such Hibernate ORM, Vert.x, Netty, RESTEasy and Apache Camel as well as the APIs from the https://microprofile.io/[Eclipse MicroProfile] initiative,
+such as https://github.com/eclipse/microprofile-config[Config] or https://github.com/eclipse/microprofile-reactive-messaging[Reactive Messaging].
+Using Quarkus, you can develop applications using both imperative and reactive styles, also combining both approaches as needed.
+
+It is designed for significantly reduced memory consumption and improved startup time.
+Last but not least, Quarkus supports both OpenJDK HotSpot and GraalVM virtual machines.
+With GraalVM it is possible to compile the application into a native binary and thus reduce the resource consumption and startup time even more.
+
+To learn more about Quarkus itself, we recommend to take a look at its excellent https://quarkus.io/get-started/[Getting Started] guide.
+
+== Consuming Kafka Messages with Quarkus
+
+In the original https://github.com/debezium/debezium-examples/tree/master/outbox[example application] demonstrating the outbox pattern,
+there was a microservice ("shipment") based on Thorntail that consumed the events produced by the Debezium connector.
+We've extended the example with a new service named "shipment-service-quarkus".
+It provides the same functionality as the "shipment-service" but is implemented as a microservice based on Quarkus instead of Thorntail.
+
+This makes the overall architecture look like so:
+
+++++
+<div class="imageblock centered-image">
+    <img src="/images/outbox_pattern_quarkus.png" style="max-width:100%; margin-bottom:20px; margin-top:20px;" class="responsive-image" alt="Outbox Pattern Overview">
+</div>
+++++
+
+To retrofit the original service into a Quarkus-based application, only a few changes were needed:
+
+ * Quarkus right now supports only MariaDB but not MySQL; hence we have included an instance of MariaDB to which the service is writing
+ * The https://javaee.github.io/jsonp/[JSON-P API] used do deserialize incoming JSON messages is not supported yet; so the code has been modified to use the Jackson API which is already is supported by Quarkus
+ * Instead of the Kafka consumer API, the https://github.com/eclipse/microprofile-reactive-messaging[Reactive Messaging API] defined by MicroProfile is used to receive messages from Apache Kafka; as an implementation of that API, the one provided by the https://github.com/smallrye/smallrye-reactive-messaging[SmallRye project] is used.
+
+While the first two steps shouldn't be needed eventually, as Quarkus gains support for more technologies and APIs over time,
```

Comment:
I'm not sure we will support MySQL. TBH, I think you should move your examples to MariaDB (well, if it were just me, it would be PostgreSQL ;)) and be done with it.

Reply:
Btw. the example uses PG, too: in the sending service. We used different ones to make the point that DBZ gives you portability across vendor boundaries.